drivers/serial/uart_ns16550.c

```diff
@@ -732,6 +735,24 @@ static int uart_ns16550_configure(const struct device *dev,
 	return ret;
 };
 
+#if defined(CONFIG_UART_NS16550_ENABLE_DEVICE_PM)
+__maybe_unused static int uart_ns16550_pm_action(const struct device *dev,
+						 enum pm_device_action action)
+{
+	struct uart_ns16550_dev_data *data = dev->data;
+	struct uart_config *uart_cfg = &data->uart_config;
+
+	switch (action) {
+	case PM_DEVICE_ACTION_RESUME:
+		return uart_ns16550_configure(dev, uart_cfg);
+	case PM_DEVICE_ACTION_SUSPEND:
+		return 0;
```

Comment:
ok so what's the story exactly? suspend does nothing, resume just reconfigures? both platforms I tried using this driver did not seem to like this at all, is it a silabs specific thing?

Reply:
Ok but then now we end up with a kconfig option that a user is expected to set and the only thing we are attaching to it is a very vague description:

```
This enables support device PM implementation for ns16550 driver.
```

that does not really help explaining, it sounds like "of course I want it wherever possible" but it's really "I need it for this class of devices"

How about having this behavior controlled by a devicetree property instead, with a proper description of what it does and when should it be enabled instead, and then you can have it set on the dts files for all silabs devices and the user would not have to worry about it.